src/test/rgw/bucket_notification/test_bn.py

```diff
@@ -1795,8 +1796,9 @@ def lifecycle(endpoint_type, conn, number_of_objects, topic_events, create_threa
     time.sleep(20)
     print('wait for sometime for the messages...')
 
-    no_keys = list(bucket.list())
     wait_for_queue_to_drain(topic_name, http_port=port)
+    time.sleep(20)
```

Comment:
does lifecycle takes more than 20 seconds?
if it does, why not wait longet in line 1796?
also, if queue is drained, it means we sent all notifications, why wait more?

Reply:
please loop over the bucket list until it ts empty. do a sleep (maybe 5s) inside the loop